catkit/hardware/iris_ao/iris_ao_controller.py

```diff
@@ -85,10 +85,13 @@ def _communicate(self, input):
             self.instrument = None  # If it died, there's nothing to close, so don't.
             raise RuntimeError(f"{self.config_id} unexpectedly closed.") from error
 
-        # Poll in case it died (though the above flush would most likely catch this).
-        #if self.instrument.poll() is not None:
-        #    self.instrument = None  # If it died, there's nothing to close, so don't.
-        #    raise RuntimeError(f"{self.config_id} unexpectedly closed: ('{self.instrument.returncode}').")
+        # Poll in case it died after receiving the above commanding.
+        if self.instrument.poll() is not None:
+            self.instrument = None  # If it died, there's nothing to close, so don't.
+            raise RuntimeError(f"{self.config_id} unexpectedly closed: ('{self.instrument.returncode}').")
+
+        # WARNING!!! - even with the above poll() there's still scope for the process to have died in this gap.
+        # In which case, the below readline() will deadlock! See https://github.com/spacetelescope/catkit/pull/189.
```

Comment:
Really, I should check whether a terminated process before ``readline()`` will not cause the ``readline()`` to fail - TBH, I would have thought that it would in the same way that we see when this happens for a flush as th pipe to ``stdout`` gets closed. If it does, then the process issue happens during the ``readline()`` and this PR does nothing 🤷  (which is why I left it out in the first place - along with the fact that even if it did do something it's still racey).